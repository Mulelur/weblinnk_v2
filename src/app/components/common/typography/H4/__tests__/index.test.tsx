import * as React from 'react';
import { render } from '@testing-library/react';
import { themes } from 'styles/theme/themes';
import { DefaultTheme } from 'styled-components';
import { H4 } from '../index';

const renderWithTheme = (theme?: DefaultTheme) =>
  render(<H4 theme={theme || themes.light} />);

describe('<H4 />', () => {
  it('should match snapshot', () => {
    const { container } = render(<H4 />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a h4 tag', () => {
    const { container } = renderWithTheme();
    expect(container.querySelector('h4')).toBeInTheDocument();
  });

  it('should have the correct theme', () => {
    const { container } = render(<H4 theme={themes.light} />);
    expect(container.firstChild).toHaveStyle(`color: ${themes.light.text}`);
  });
});
