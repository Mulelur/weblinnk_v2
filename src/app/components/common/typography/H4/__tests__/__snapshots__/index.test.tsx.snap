// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H4 /> should match snapshot 1`] = `
.c0 {
  font-weight: 500;
}

<h4
  class="c0"
/>
`;
